src/gui/systray.cpp

```diff
@@ -144,123 +136,88 @@ void Systray::setToolTip(const QString &tip)
+    for (int i = 0; i < QGuiApplication::screens().count(); i++) {
+        if (QGuiApplication::screens().at(i)->geometry().contains(qPos)) {
+            return i;
         }
     }
-#endif
+    return 0;
 }
-int Systray::calcTrayWindowY()
-{
-#ifdef Q_OS_OSX
-    // macOS menu bar is always 22 (effective) pixels
-    // don't use availableGeometry() here, because this also excludes the dock
-    return 22+6;
-#else
-    QScreen* trayScreen = nullptr;
-    if (QGuiApplication::screens().count() > 1) {
-        trayScreen = QGuiApplication::screens().at(screenIndex());
-    } else {
-        trayScreen = QGuiApplication::primaryScreen();
-    }
-
-    int screenHeight = trayScreen->geometry().height();
-    int availableHeight = trayScreen->availableGeometry().height();
-
-    QPoint topRightDpiAware = QPoint();
-    QPoint topLeftDpiAware = QPoint();
-    if (this->geometry().left() == 0 || this->geometry().top() == 0) {
-        // tray geometry is invalid - QT bug on some linux desktop environments
-        // Use mouse position instead. Cringy, but should work for now
-        topRightDpiAware = QCursor::pos() / trayScreen->devicePixelRatio();
-        topLeftDpiAware = QCursor::pos() / trayScreen->devicePixelRatio();
-    } else {
-        topRightDpiAware = this->geometry().topRight() / trayScreen->devicePixelRatio();
-        topLeftDpiAware = this->geometry().topLeft() / trayScreen->devicePixelRatio();
-    }
-    // get y coordinate from top center point of tray icon
-    int trayIconTopCenterY = (topRightDpiAware - ((topRightDpiAware - topLeftDpiAware) * 0.5)).y();
 
-    if (availableHeight < screenHeight) {
-        // taskbar is on top or bottom
-        if (trayScreen->availableGeometry().y() > trayScreen->geometry().y()) {
-            // on top
-            return (screenHeight - availableHeight) + 6;
-        } else {
-            // on bottom
-            return screenHeight - 510 - (screenHeight - availableHeight) - 6;
-        }
-    } else {
-        // on the left or right
-        return (trayIconTopCenterY - 510 + 12);
+int Systray::taskbarOrientation()
+{
+// macOS: Always on top
+#if defined(Q_OS_MACOS)
+    return TaskBarPosition::Top;
+// Windows: Check registry for actual taskbar orientation
+#elif defined(Q_OS_WIN)
+    auto taskbarGeometry = Utility::registryGetKeyValue(HKEY_CURRENT_USER,
+        "SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Explorer\\StuckRects3",
+        "Settings");
+    switch (taskbarGeometry.toInt()) {
+    // Mapping windows binary value (0 = left, 1 = top, 2 = right, 3 = bottom) to qml logic (0 = bottom, 1 = left...)
+    case 0:
+        return TaskBarPosition::Left;
+    case 1:
+        return TaskBarPosition::Top;
+    case 2:
+        return TaskBarPosition::Right;
+    case 3:
+        return TaskBarPosition::Bottom;
+    default:
+        return TaskBarPosition::Bottom;
     }
+// Else (generally linux DEs): fallback to cursor position nearest edge logic
+#else
+    return 0;
```

Comment:
You meant TaskBarPosition::Bottom here I guess?

Reply:
Indeed.